GitCommands/Git/Executable.cs

```diff
@@ -232,6 +233,9 @@ public StreamReader StandardError
             /// <inheritdoc />
             public Task<int> WaitForExitAsync() => _exitTaskCompletionSource.Task;
 
+            /// <inheritdoc />
+            public Task WaitForExitAsync(CancellationToken token) => _process.WaitForExitAsync(token);
```

Comment:
Can this return the exit code, for parity with sibling methods?

Also, why is this method implemented so differently to the overload that does not support cancellation?

Reply:
It is likely possible to get exit status etc from the returned task, but exceptions must be handled in the same way the exitTask handles it now.

To be able to cancel the process, the process must be involved. The other interfaces are add-on to return data from the process.
This method is really something different than the overload without cancellation. Maybe the existing method should be renamed to `WaitForExitTaskAsync` to clarify this.

Maybe there are entirely better ways to do this?